Automattic/jetpack
# Use spread operator instead of func_get_args

PHP 5.6 supports the spread operator (`...$var`) for variadic functions. As an language construct, it is faster than using `func_get_args`. As a minor tweak, this PR upgrades instances where the function had no declared arguments and fully depended on `func_get_args` when it wanted to process variadic arguments.

Intentionally not including what is already pending in https://github.com/Automattic/jetpack/pull/13695

#### Changes proposed in this Pull Request:
* Replace instances of `func_get_args` with the spread operator.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* n/a

#### Testing instructions:
* Run unit tests (some changes involved the unit testing structure for JITM)
* Interact with WP.com and ensure everything works as expected (XMLRPC)
* Add posts, change options, etc, and verify those changes synced to WP.com.

#### Proposed changelog entry for your changes:
* Performance: Modernize various parts of the code to use PHP 5.6+ functionality.
